fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -102,11 +102,14 @@ public OptExpression optimize(ConnectContext connectContext,
 
         ruleRewriteOnlyOnce(memo, rootTaskContext, RuleSetType.PARTITION_PRUNE);
         ruleRewriteIterative(memo, rootTaskContext, new PruneProjectRule());
-        ruleRewriteIterative(memo, rootTaskContext, new ScalarOperatorsReuseRule());
         ruleRewriteOnlyOnce(memo, rootTaskContext, new JoinForceLimitRule());
+        ruleRewriteIterative(memo, rootTaskContext, new MergeProjectWithChildRule());
+        ruleRewriteOnlyOnce(memo, rootTaskContext, new ScalarOperatorsReuseRule());
```

Comment:
```suggestion
        ruleRewriteOnlyOnce(memo, rootTaskContext, new ScalarOperatorsReuseRule());
        ruleRewriteIterative(memo, rootTaskContext, new MergeProjectWithChildRule());
```

Reply:
Need to merge Project first, and then perform expression reuse

Reply:
Why don't run ReuseRule first? I think you don't need care it